Documentation for Gemini G4V controller (#359)
mixxxdj/manual: source/hardware/controllers/gemini_g4v.rst

Gemini G4V
=====================

> .. figure:: ../../_static/controllers/gemini_g4v_topview.jpg
>    :align: center
>    :width: 100%
>    :figwidth: 100%
>    :alt: Behringer CMD Micro
>    :figclass: pretty-figures
> 

Reviewer: ```suggestion
```

Please remove, the other controller pages don't show photos of the controller either.

They are not accessible nor controlled by Mixx so they will not appear in any recording nor is possible to control them via the Mixxx screen interface.

-  `Manufacturer’s product page <http://geminisound.com/product/g4v>`__
-  `Forum thread <https://www.mixxx.org/forums/viewtopic.php?f=6&t=12919>`__
-  `Manufacturer’s User Guide <https://www.manualslib.com/manual/826563/Gemini-G4v.html>`__

.. versionadded:: 2.3

Mixxx sound hardware preferences
--------------------------------

========== =============
Output     Channel
========== =============
Master     Channel 1 - 2
Headphones Channel 3 - 4
========== =============

Mapping Description
-------------------

Most of the board's controls work as described in the G4V's manual, with a few exceptions. Snapshots of the board's labeling from the Manual are included below:

.. figure:: ../../_static/controllers/gemini_g4v_controls.png
   :align: center
   :width: 100%
   :figwidth: 100%
   :alt: Behringer CMD Micro
   :figclass: pretty-figures

.. figure:: ../../_static/controllers/gemini_g4v_frontview.jpg
   :align: center
   :width: 100%
   :figwidth: 100%
   :alt: Behringer CMD Micro
   :figclass: pretty-figures

> 

Reviewer: These should be vector graphics if possible. The labels are blurry and barely readable, and raster images are hard to edit.
Use can use these: [gemini_g4v.zip](https://github.com/mixxxdj/manual/files/6164503/gemini_g4v.zip)

I also suggest make separate sections for top, front and rear and make the sections below subsections of the section that the control belongs to.

Reviewer: Also, please fix the `:alt:` label ;-)

.. figure:: ../../_static/controllers/gemini_g4v_controllist.jpg
   :align: center
   :width: 100%
   :figwidth: 100%
   :alt: Behringer CMD Micro

>    :figclass: pretty-figures

Reviewer: This should be text, so it can be read out by a screen reader.

I suggest to use multiple markdown tables, one for each section (top, front, rear).

Author: > This should be text, so it can be read out by a screen reader.

Sorry. I don't understand that one. What should be text?

Author: OK. Now I got it.
Anyway, not sure how useful is to have that in text, this list is to identify the different controls in the drawing. Any person with impaired vision will have problems to use that information in the images.

Reviewer: But it's not searchable with CTRL+F. And rasterized images have bad quality, especially if it's resized for display.

It's also hard to find the control you're looking for, because all controls are in a single list. Using different subsections for top, front, rear would be helpful: Have a look at this manual page for reference: https://manual.mixxx.org/2.3/en/hardware/controllers/roland_dj_505.html#deck-section-p-5-6

.. figure:: ../../_static/controllers/gemini_g4v_controls.png
   :align: center
   :width: 100%
   :figwidth: 100%
   :alt: Behringer CMD Micro
   :figclass: pretty-figures

.. figure:: ../../_static/controllers/gemini_g4v_frontview.jpg
   :align: center
   :width: 100%
   :figwidth: 100%
   :alt: Behringer CMD Micro
   :figclass: pretty-figures

.. figure:: ../../_static/controllers/gemini_g4v_controllist.jpg
   :align: center
   :width: 100%
   :figwidth: 100%
   :alt: Behringer CMD Micro
   :figclass: pretty-figures

Shift
-----

On each of the physical decks is a :hwlabel:`SHIFT` button.

This button, when held, changes the behavior of various buttons on that deck, labeled under the respective button.

Mixer
-----

In the middle of board, there are :hwlabel:`HIGH`, :hwlabel:`MID`, and :hwlabel:`LOW` equalizer knobs, :hwlabel:`GAIN`, and :hwlabel:`FILTER` knobs for the individual virtual decks that work as normal.

The :hwlabel:`MASTER` and :hwlabel:`CUE MIX` knobs works as usual.

The :hwlabel:`BOOTH` knob controls the gain of the booth audio output. This output is managed internally by the controller.

Crossfader
----------

> The :hwlabel:`CROSSFADER` works as usual.

Reviewer: ```suggestion
The Crossfader works as usual.
```

It's not labeled on the controller, right?

Author: You're correct, is not labelled in the controller, only in the documentation.

Reviewer: Yes.

The numbered buttons on the sides of the crossfader change the orientation of the virtual decks with respect to the crossfader.

Deck Select Buttons
-------------------

> Just below the Tempo slider on either side of the board is a :hwlabel:`Deck Sel` button (1|3 on the left and 2|4 on the right).

Reviewer: ```suggestion
Just below the Tempo slider on either side of the board is a :hwlabel:`DECK SEL` button (:hwlabel:`C|A` on the left and :hwlabel:`B|D` on the right).
```

Author: Why to change 1|3 (the actual label in the controller) to C|A ?

Reviewer: Sorry, I used a wring image of thd controller for reference: https://idmmag.com/wp-content/uploads/2013/01/Gemini-G4V.jpg

Now I see that thf label is different on the actual controller.

Author: Ouch. Yes, this is really confusing.
Don't worry. I contested to the only two that where really relevant.
I appreciate your comments and agree that they increase readability. Any other comment will be welcomed.
I think that now the quality is quite good.
Is there anything else on my side to be done?
Thanks.
FYI. I didn't pay too much attention to the case because the hwlabel tag already makes the text to become upper case in the final document.

Reviewer: I'll review again using the correct labels. Maybe we need to revert some of my proposed changes, sorry.

Reading my comment from above again, I feel it's necessary to point that I was not drunk when I commented, it was just late and I wrote the comment my phone :D

Jog Wheels
----------

Pressing the top of the Jog Wheel will activate scratching mode mode. It will stop playing.

Turning the jog wheel will move the track play head, for scratching or fine placement.

Turning the jog wheel while pressing shift will change the track play head fast, for fast search.

When the jog wheel top is released, the jog wheel will stay in scratching mode until it stops moving, for spin backs or similar actions

If the deck was in slip mode before entering in scratching mode, the play head will move to the position it will have been if no manipulation happened, useful for scratching without losing the beat

Turning the jog wheel without pressing the top will temporarily change the tempo, for nudging.

Performance Pads
----------------

Above the wheel on each deck is a set of 8 performance pads (top pads numbered 1-4 and bottom pads numbered 5-8 for the remainder of this manual page).

The functionality of the performance pads is determined by the pads mode: Hot Cue, Auto Loop, Sample, Loop Roll, manual Loop and Beat Jump.

You can can change the mode by using the mode buttons as indicated.

Hot Cue Mode
^^^^^^^^^^^^

> To enable Hot Cue mode, press the :hwlabel:`Hot Cue` button.

Reviewer: ```suggestion
To enable Hot Cue mode, press the :hwlabel:`HOT CUE` button.
```

When the performance pads are in Hot Cue mode, the Hot Cue button will be lit up.

Pressing a performance pad when in Hot Cue mode will control the corresponding numbered Hot Cue (Performance Pad 1 controls Hot Cue 1, etc) on the virtual deck.
If the corresponding Hot Cue isn't set, pressing the performance pad will set a hot cue at that point.
If the corresponding Hot Cue is set, pressing the performance pad will move playback to that hot cue.
If a hot cue is set and it's corresponding performance pad is pressed while the deck's Shift button is being held, the hot cue will be deleted.

Sample Mode
^^^^^^^^^^^

> To enable Sample mode, press the :hwlabel:`Sample` button.

Reviewer: ```suggestion
To enable Sample mode, press the :hwlabel:`SAMPLE` button.
```

When the performance pads are in Sample mode, the Sample button will be lit up.
It will show the sample decks if the skin has support for them.

Pads will be lit up if they have a track loaded.
Each deck control 8 samples, deck 1 controls samples 1 to 8, deck 2 controls samples 9 to 16, an so on.

The :hwlabel:`Sample Volume` knobs control the volume for all the samples.

Auto Loop Mode
^^^^^^^^^^^^^^

> To enable looping mode, press the :hwlabel:`Auto Loop` button.

Reviewer: ```suggestion
To enable looping mode, press the :hwlabel:`AUTO LOOP` button.
```

When the performance pads are in Auto Loop mode, the Auto Loop button will be lit up.

Pressing a pad starts a loop of the length indicated in the pad, the pad will lit up.
Pressing the lit pad stop the loop.

> Pressing a different pad will stop the first pad and will Cstart a new loop.

Reviewer: ```suggestion
Pressing a different pad will stop the current loop and start a new one.
```

> The :hwlabel:`Loop Move` knob shifts the loop position in the track.

Reviewer: ```suggestion
The :hwlabel:`LOOP MOVE` knob shifts the loop position in the track.
```

Loop Roll Mode
^^^^^^^^^^^^^^

> To enable loop roll mode, press the :hwlabel:`Loop Roll` button.

Reviewer: I don't see a LOOP ROLL knob labeled on the controller.

Author: It's the rightmost button under the pads.

Reviewer: Okay, found it:https://i.pinimg.com/originals/1c/e0/c2/1ce0c2af2d3f455ca0cbca88c384c17a.jpg

Before, I used this image which has different labels for some reason. In that case some of my changes are probably wrong:
https://idmmag.com/wp-content/uploads/2013/01/Gemini-G4V.jpg

When the performance pads are in Loop Roll mode, the Loop Roll button will be lit up

Pressing a pad start a loop of the length indicated in the pad, the pad will lit up.
Releasing the pad will end the loop and the track position will move to the position it will have been if the loop never happened.

> The:hwlabel:`Loop Move` knob shifts the loop position in the virtual deck.

Reviewer: ```suggestion
The :hwlabel:`LOOP MOVE` knob shifts the loop position in the virtual deck.
```

Manual Loop Mode
^^^^^^^^^^^^^^^^

> To enable Manual Loop mode, press the :hwlabel:`Loop Roll` button while holding the Shift button.
> When the performance pads are in Manual Loop mode, the :hwlabel:`Loop Roll` button will flash.

Reviewer: ```suggestion
To enable Manual Loop mode, press the :hwlabel:`LOOP ROLL` button while holding the Shift button.
When the performance pads are in Manual Loop mode, the :hwlabel:`LOOP ROLL` button will flash.
```

When a manual loop is set, all 8 performance pads are lit.

> Performance pad 1 sets the loop IN point.
> 
> Performance pad 2 sets the loop OUT point.
> 
> Performance pads 3 enables and disables the loop once the IN and OUT points are set.

Reviewer: ```suggestion
Performance pad 1 sets the loop in point.

Performance pad 2 sets the loop out point.

Performance pads 3 enables and disables the loop once the in and out points are set.
```

Performance pad 5 doubles the length of the loop.

Performance pad 6 halves the length of the loop.

Performance pad 7 moves the loop 1 beat backward.

Performance pad 8 moves the loop 1 beat forward.

Beat Jump Mode
^^^^^^^^^^^^^^

> To enable Beat Jump mode, press the :hwlabel:`Sample`  button while holding the :hwlabel:`Shift` button.

Reviewer: ```suggestion
To enable Beat Jump mode, press the :hwlabel:`SAMPLE`  button while holding the :hwlabel:`SHIFT` button.
```

> When the performance pads are in Beat Jump mode, the :hwlabel:`Sample` button will flash

Reviewer: ```suggestion
When the performance pads are in Beat Jump mode, the :hwlabel:`SAMPLE` button will flash
```

Pressing a pad jumps forward the number of beats indicated in the pad.

> Pressing a pad while holding the :hwlabel:`Shift` button jumps backward the number of beats indicated in the pad.

Reviewer: ```suggestion
Pressing a pad while holding the :hwlabel:`SHIFT` button jumps backward the number of beats indicated in the pad.
```

FX and Effects
--------------

Each virtual deck has assigned 1 Effect Unit, with the deck number being the same as the Effect Unit number.

Pressing the deck's :hwlabel:`FX ON/OFF` button will enable/disable the Effect Unit corresponding to the virtual deck number.

The :hwlabel:`FX ON/OFF` button is lit when the virtual deck's corresponding Effect Unit is enabled.

> Holding :hwlabel:`Shift` and pressing the :hwlabel:`FX Select` button will scroll through the Effect Unit's effect chains.

Reviewer: ```suggestion
Holding :hwlabel:`SHIFT` and pressing the :hwlabel:`FX SEL` button will scroll through the Effect Unit's effect chains.
```

> The :hwlabel:`Par 1` and :hwlabel:`Par 2` knobs control the meta knob and the effect mix level, respectively, for the Effect Chain.

Reviewer: ```suggestion
The :hwlabel:`PAR 1` and :hwlabel:`PAR 2` knobs control the meta knob and the effect mix level, respectively, for the Effect Chain.
```

Slip Mode
---------

> The :hwlabel:`Slip` button enables/disables slip mode on the virtual deck, the button will be lit up if Slip mode is on.

Reviewer: ```suggestion
The :hwlabel:`SLIP` button enables/disables slip mode on the virtual deck, the button will be lit up if Slip mode is on.
```

Misc Controls
-------------

>     The :hwlabel:`Keylock` button enables/disables keylock on the virtual deck.

Reviewer: ```suggestion
The :hwlabel:`KEYLOCK` button enables/disables keylock on the virtual deck.
```

> Unused controls
> ---------------
> 
> Below is a list of controls on the board that currently do not have any functionality.
> 
> The :hwlabel:`Bank Up/Bank Down` button.
> 
> The :hwlabel:`Scratch` button.

Reviewer: ```suggestion
Known Issues
------------

- The :hwlabel:`BANK UP`/:hwlabel:`BANK DOWN` buttons are not mapped.
- The :hwlabel:`SCRATCH` button is not mapped.
```

To enable Hot Cue mode, press the :hwlabel:`HOT CUE` button.
When the performance pads are in Hot Cue mode, the Hot Cue button will be lit up.

> Pressing a performance pad when in Hot Cue mode will control the corresponding numbered Hot Cue (:hwlabel:`Pad 1` controls Hot Cue 1, etc) on the virtual deck.

Reviewer: This is Pad actually labeled as "Pad 1" on the controller? If not, don't use `:hwlabel:` here.

When a manual loop is set, all 8 performance pads are lit.

================ ===========================================================================
Pad                Function
================ ===========================================================================
:hwlabel:`pad 1` Sets the loop in point.

:hwlabel:`pad 2` Sets the loop out point.

:hwlabel:`pad 3` Enables and disables the loop once the in and out points are set.

:hwlabel:`pad 5` Doubles the length of the loop.

:hwlabel:`pad 6` Halves the length of the loop.

:hwlabel:`pad 7` Moves the loop 1 beat backward.

> :hwlabel:`pad 8` Moves the loop 1 beat forward.

Reviewer: same here, don't use `:hwlabel:` if there is no label on the controller.
